daml/Marketplace/Issuance/AssetDescription.daml

```diff
@@ -44,6 +50,27 @@ template AssetDescription
     key assetId : Id
     maintainer key.signatories
 
+    nonconsuming choice Expiry : Optional Date
+      with
+        party : Party
+      controller party
+      do
+        return $ expiry (C.deserialize claims)
+
+    nonconsuming choice Underlying : [Id]
+      with
+        party : Party
+      controller party
+        do
+          return $ snd <$> payoffs (C.deserialize claims)
+
+    nonconsuming choice Multipliers : [Observation Date Decimal]
```

Comment:
hmm, I know I suggested you change this in the first place, but now I'm worried that the `Observation` return type may be confusing. You have to at least let the client know that this needs to be pattern matched with `[Const multiplier]` (i.e. singleton list with a `Const`) for futures; I think otherwise they will be confused.